production/bin/create_configs.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# generate configurartion files with group and world read permissions.
+umask 022
+
+echo "Starting application configuration. Processing ENV settings."
+
+# Move default configs into place.
+# For most environments, these settings are overridden in ENV set from SSM.
+(
+  cd config
```

Comment:
If we are only using this config generation script in production, I wonder if we should be copying over many of these files? I worry about using development-related configuration (like in config.yml, since we have default values for all environments; or cookies.txt) accidentally rather than bailing on the deploy.

Reply:
My understanding is that we'll be using the default configs for unit tests and integration tests. In these situations, we want to use the default configs as they are sufficient for testing, and we need to move them into place if they don't exist for the application to start under test.

The concern about starting with a default configuration in Live is indeed a concern - typically when I make changes like this I will watch the deploy in ECS Events and follow the CloudWatch logs to confirm things start as expected, and configs are updated from SSM as intended. Since we do this migration once (moving from configurator to SSM driven configuration) I think the risk is manageable, but maybe we can make this more robust like you mention, where Live deploys fail rather than use defaults.

Reply:
If we can confirm that everything is working well to start then that sounds good for the migration, especially since we'll want this for the cookies.txt changes that are already on develop and about to be deployed!

I'd like to get these changes made if we can, though, since I worry about someone forgetting to set a new config value going forward and defaulting to a dev/test value. We could have copying example files be run as part of the test run on CI but not a part of the start step, I think.